src/Encamina.Enmarcha.SemanticKernel/MemoryManager.cs

```diff
@@ -26,12 +27,22 @@ public class MemoryManager : IMemoryManager
     /// Initializes a new instance of the <see cref="MemoryManager"/> class.
     /// </summary>
     /// <param name="memoryStore">A valid instance of a <see cref="IMemoryStore"/> to manage.</param>
-    public MemoryManager(IMemoryStore memoryStore, ILogger<MemoryManager> logger)
+    /// <param name="memoryStorageEventHandlerDelegate">A delegate to handle events from <see cref="MemoryStorageEvent"/>.</param>
+    /// <param name="logger">Log service.</param>
+    public MemoryManager(IMemoryStore memoryStore, ILogger<MemoryManager> logger, Action<string, MemoryStorageEventArgs> memoryStorageEventHandlerDelegate)
```

Comment:
People do not use the `delegate` anymore. Just call it `memoryStorageEventHandler`. The fact that it is an `Action<>` already indicates it is a delegate.